### PR TITLE
test: Migrate vm-download to use virt-builder

### PR DESCRIPTION
**Pull request by stefwalter**
This lets us use virt-builder indexes.

### CODE REVIEW

File: test/vm-download [hunk 1]
```diff
@@ -18,78 +18,45 @@
...
+    key = args.flavor + "-" + system + "-" + revision
+
+if not os.path.exists(cache):
+    os.makedirs(cache)
+
+cmd = [
+    "virt-builder",
+    "--cache", cache,
+    "--no-check-signature", # Unfortunately, not ready for this yet
+    "--arch=x86_64",
```
Comment (mvollmer) [inline test/vm-download]: x86_64 probably shouldn't be hardocded here.

File: test/vm-download [hunk 2]
```diff
@@ -18,78 +18,45 @@
...
+    "--format=qcow2",
+    "--output", os.path.join(images, image),
+    "--root-password=password:foobar",
```
Comment (mvollmer) [inline test/vm-download]: The images already have the right root password, no?

Comment (stefwalter) [inline test/vm-download]: If you don't put this in then virt-builder randomizes the root password.

File: test/vm-download (hunk 1 above)
Comment (stefwalter) [inline test/vm-download]: Fixed

File: test/vm-download [hunk 3]
```diff
@@ -18,78 +18,45 @@
...
-        exit 0
-    fi
-fi
-
-if [ -f "$checksum_file" ] && cmp "$checksum_file.remote" "$checksum_file"; then
-    rm -f "$checksum_file.remote"
-    exit 0
-fi
-
-cat "$checksum_file.remote" | while read sum file; do
```
Comment (mvollmer) [inline test/vm-download]: The checksum file (which is really a manifest) can contain all kinds of files in addition to the QEMU image.  For Atomic we also have a ISO for cloud-init, for example.

**Comment by mvollmer** [comment]
I think this will double the local storage use, no?  We'll have the ready-to-use image in the cache, and then virt-builder makes a  copy of it, adjusting the size, and tweaking it a bit, both of which shouldn't be necessary.

(I haven't tried because I have crappy network here right now.)

File: test/vm-download (hunk 2 above)
Comment (mvollmer) [inline test/vm-download]: I see... isn't this an indication that virt-builder is doing too much, more than we want from it?

**Comment by stefwalter** [comment]
> I think this will double the local storage use, no? We'll have the ready-to-use image in the cache, and then virt-builder makes a copy of it, adjusting the size, and tweaking it a bit, both of which shouldn't be necessary.

It does store the xz version in a cache, so no not double by a long shot:

```
 * xz version in cache: 822,954,436
 * uncompressed version: 2,368,471,040
```

But if this is not acceptable, we could skip the cache and just use --no-cache. We would then keep two sets of checksum logic for both the compressed and uncompressed versions. I would rather not keep the later checksum logic.

**Comment by stefwalter** [comment]
> But if this is not acceptable, we could skip the cache and just use --no-cache. We would then keep two sets of checksum logic for both the compressed and uncompressed versions. I would rather not keep the later checksum logic.

Alternatively, we could assume that virt-builder only places the resulting image download right, and if an image exists in the correct location/name then we don't have to download it again. I'll do this.

**Comment by mvollmer** [comment]
What about the ISO files for Atomic images? Should we try to get rid of them?

**Comment by dperpeet** [comment]
> What about the ISO files for Atomic images? Should we try to get rid of them?

If we can properly initialize the images without cloud-init and thus get rid of that step, the test-framework code would be cleaner. So if it's doable without much effort, we should get rid of the iso boot files.

**Comment by stefwalter** [comment]
Yes, I/we should fix the ISO issue before merging this pull request.

**Comment by mvollmer** [comment]
@dperpeet, @stefwalter, either this or #2822 should be closed, no?

**Comment by stefwalter** [comment]
Replaced by #3037

**Pull request closed without merging**